scipy/stats/_stats_py.py

```diff
@@ -9865,6 +9877,98 @@ def _cdf_distance(p, u_values, v_values, u_weights=None, v_weights=None):
     return np.power(np_vecdot(np.power(np.abs(u_cdf - v_cdf), p), deltas), 1/p)
 
 
+def _cdf_distance_iv(x, y, wx, wy, axis):
+    xp = array_namespace(x, y, wx, wy)
+    x, y, wx, wy = xp_promote(x, y, wx, wy, force_floating=True, xp=xp)
+
+    x, wx = xp.broadcast_arrays(x, wx) if wx is not None else (x, wx)
+    y, wy = xp.broadcast_arrays(y, wy) if wy is not None else (y, wy)
+
+    if axis is None:
+        x, y = xp_ravel(x), xp_ravel(y)
+        wx = wx if wx is None else xp_ravel(wx)
+        wy = wy if wy is None else xp_ravel(wy)
+        axis = -1
+
+    if wx is None and wy is None:
+        x, y = _broadcast_arrays((x, y), axis=axis, xp=xp)
+        x, y, = xp.moveaxis(x, axis, -1), xp.moveaxis(y, axis, -1)
+    elif wx is None:
+        x, y, wy = _broadcast_arrays((x, y, wy), axis=axis, xp=xp)
+        x, y, wy = (xp.moveaxis(x, axis, -1), xp.moveaxis(y, axis, -1),
+                    xp.moveaxis(wy, axis, -1))
+    elif wy is None:
+        x, y, wx = _broadcast_arrays((x, y, wx), axis=axis, xp=xp)
+        x, y, wx = (xp.moveaxis(x, axis, -1), xp.moveaxis(y, axis, -1),
+                    xp.moveaxis(wx, axis, -1))
+    else:
+        x, y, wx, wy = _broadcast_arrays((x, y, wx, wy), axis=axis, xp=xp)
+        x, y, wx, wy = (xp.moveaxis(x, axis, -1), xp.moveaxis(y, axis, -1),
+                        xp.moveaxis(wx, axis, -1), xp.moveaxis(wy, axis, -1))
+
+    # technically, if only one of wx were infinite along `axis`, we could
+    # treat it as if that weight were unity and all the rest were zero.
+    wx = wx if wx is None else xp.where((wx < 0) | xp.isinf(wx), xp.nan, wx)
+    wy = wy if wy is None else xp.where((wy < 0) | xp.isinf(wy), xp.nan, wy)
+
+    return x, y, wx, wy, axis, xp
+
+
+def _cdf_distance(p, x, y, wx=None, wy=None, axis=0):
+    # vectorized version of _cdf_distance_ref.
```

Comment:
If this is hard to review, I will submit a PR with a vectorized `xp_searchsorted`, and we can try using that to more directly translate `_cdf_distance_reference`.